app/src/main/java/org/dhis2/usescases/datasets/datasetDetail/datasetList/mapper/DatasetCardMapper.kt

```diff
@@ -0,0 +1,234 @@
+    val resourceManager: ResourceManager,
+) {
+
+    fun map(
+        dataset: DataSetDetailModel,
+        onSyncIconClick: () -> Unit,
+        onCardCLick: () -> Unit,
+    ): ListCardUiModel {
+        return ListCardUiModel(
+            title = dataset.namePeriod(),
+            lastUpdated = dataset.lastUpdated().toDateSpan(context),
+            additionalInfo = getAdditionalInfoList(dataset),
+            actionButton = {
+                ProvideSyncButton(
+                    state = dataset.state(),
+                    onSyncIconClick = onSyncIconClick,
+                )
+            },
+            expandLabelText = resourceManager.getString(R.string.show_more),
+            shrinkLabelText = resourceManager.getString(R.string.show_less),
+            onCardCLick = onCardCLick,
+        )
+    }
+
+    private fun getAdditionalInfoList(
+        dataset: DataSetDetailModel,
+    ): List<AdditionalInfoItem> {
+        val list = mutableListOf<AdditionalInfoItem>()
+
+        checkRegisteredIn(
+            list = list,
+            dataset = dataset,
+        )
+
+        checkCategoryCombination(
+            list = list,
+            dataset = dataset,
+        )
+
+        checkDatasetCompletion(
+            list = list,
+            completed = dataset.isComplete,
+        )
+
+        checkSyncStatus(
+            list = list,
+            state = dataset.state(),
+        )
+
+        // checkViewOnly()
+
+        return list
+    }
+
+    private fun checkCategoryCombination(
+        list: MutableList<AdditionalInfoItem>,
+        dataset: DataSetDetailModel,
+    ) {
+        if (dataset.nameCategoryOptionCombo() != "default") {
+            list.add(
+                AdditionalInfoItem(
+                    key = "${dataset.nameCategoryOptionCombo()}:",
+                    value = dataset.nameCatCombo(),
+                    isConstantItem = true,
+                ),
+            )
+        }
+    }
+
+    private fun checkSyncStatus(
+        list: MutableList<AdditionalInfoItem>,
+        state: State,
+    ) {
+        val item = when (state) {
+            State.TO_POST,
+            State.TO_UPDATE,
+            -> {
+                AdditionalInfoItem(
+                    icon = {
+                        Icon(
+                            imageVector = Icons.Outlined.SyncDisabled,
+                            contentDescription = resourceManager.getString(R.string.not_synced),
+                            tint = AdditionalInfoItemColor.DISABLED.color,
+                        )
+                    },
+                    value = resourceManager.getString(R.string.not_synced),
+                    color = AdditionalInfoItemColor.DISABLED.color,
+                    isConstantItem = true,
+                )
+            }
+
+            State.UPLOADING -> {
+                AdditionalInfoItem(
+                    icon = {
+                        ProgressIndicator(type = ProgressIndicatorType.CIRCULAR)
```

Comment:
Verify this with Marcos but I believe he wanted us to use the same Icon as the sync button, because of the problem with the width of the progress indicator at that size

Reply:
Yes, I saw it and we talked, I forgot to change it.  I'll change it, thanks!.